### ISSUE TITLE
Show Download list in separate Screen

### ISSUE THREAD

**Issue by gouri-panda** [kiwix/kiwix-android#2496]
**Is your feature request related to a problem? Please describe.**

It would be ideal if we show users a separate screen for downloading content(Zim files). Therefore they will know what content are they downloading on one screen and they can easily pause or stop the downloading content.

**Describe the solution you'd like**
It will be much better if we add a new screen for downloading content.

**Describe alternatives you've considered**
We show all the downloading contents in the Library fragment instead of the new screen but that will make more complicate in the fragment section of the app.

**Comment by kelson42** [comment]
It was before in a separate screen. I think the current setup is better and it was requested.
I would like to know which problem this ticket is suppose to fix... Or why we need an improvement?

**Comment by gouri-panda** [comment]
Now if a user downloading multiple files if he wants to cancel a downloading file then he has to scroll the list where he found the item also we can save the download list and if the user wants to download a file again we can easily download for them.

**Comment by kelson42** [comment]
I agree with the probelm description. But I believe having a section gathering current downloads at the top is a far better idea to solve it.

**Comment by gouri-panda** [comment]
>I believe having a section gathering current downloads at the top is a far better idea to solve it.
Yes if we can show the download list on the download fragments top then that would be awesome!

**Comment by macgills** [comment]
I think we will want to move active downloads to library, when thinking about how the app will work with the OPDS endpoint I think it is the least confusing as you could start searching for something separate from the file you just clicked download on and with this solution it would still be present. Doesn't feel very tidy to me but I could be convinced

**Comment by abdulwd** [comment]
As Emmanuel said, it was in a separate screen before. Personally, I prefer the current UI.

**Comment by Mitia99** [comment]
I hope that someone would work on this issue cause viewing multiples downloads progression from notifications is not really convenient(especially when user has tons of notifications).

**Comment by rishujam** [comment]
Hi, I just started understanding the code. Are we talking about the download that happens for the first time
<img width="215" alt="Screenshot 2023-01-19 at 12 52 00 AM" src="https://user-images.githubusercontent.com/74773876/213274943-a5331f60-673a-4f48-a9c6-0cccceb1690c.png">
If not please tell which downloads we are talking here

**Comment by kelson42** [comment]
Your screenshot is a custom app, a special flavour of the Kiwix app. I suspect the answer to that problem to be in the Readme or controbuting files.


@MohitMaliDeveloper gouri-panda how is that possible that default configuration lead developer to compile a custom app in place of kiwix app?

**Comment by kelson42** [comment]
@gouri-panda Glad if the default config is OK, still strange this happens that often!

**Comment by gouri-panda** [comment]
@kelson42 it's not problem of ours.  I think developers may have changed their default configuration. It's up to Android studio which module to run in default. After the first run, Android studio automatically chooses the last one in default.

**Comment by kelson42** [comment]
@gouri-panda hmmm, I had this discussion years ago with @mhutti1. Now I remember. Sure Android Studio has not a way meanwhile to specify the default option?

**Comment by gouri-panda** [comment]
@kelson42 Ok, I'll look into it. Ofcourse Intelij idea  may have some options for it if we wanna customize.

**Comment by kingbass01** [comment]
Hi, I would like to work on this issue but I have some doubts : 

1. Do we have to add **current downloads** to the same view as **already downloaded** or should I create another view above the already downloaded view?
2. From which module/file can I access the list containing the current downloads?
3. Do the cards have to be interactive with a progress bar showing how much percentage download has been completed? 
Thanks

**Comment by rishujam** [comment]
Is this project not active?

**Comment by kingbass01** [comment]
> Hi, I would like to work on this issue but I have some doubts : 
> 
> 1. Do we have to add current downloads to the same view as already downloaded or should I create another view above the already downloaded view?
> 2. From which module/file can I access the list containing the current downloads?
> 3. Do the cards have to be interactive with a progress bar showing how much percentage download has been completed? 
> 
> Thanks

@gouri-panda Can you help?

**Comment by kingbass01** [comment]
@kelson42 can you help?

**Comment by kelson42** [comment]
@gouri-panda @MohitMaliDeveloper I have reread the whole ticket and I believe before anybody starts to works on this, we ahould ensure to agree on what should be implemented.

There was IMO a kind of consensus to display the downloads in online library, sticked on the top. But @macgills with his comment has actually pointed importants things. On the top of this, I woukd add that I'm not happy either with current behaviour where user monitor download progress of a content (in online library) and when its over the item just disappears... so he can't open it without manually moving back to local library. This is not user friendly.

Following @macgills proposal and putting current downloads at the top of the local library seems to me the best approach... and to you?

**Comment by kelson42** [comment]
> @kelson42 can you help?

Hey, this is not a programming course here. If you are not able to find out by yourself which module is responsible for what, then you probably better leave it. We provide help on very specific questions which might be hard to answer because very technical or because its about defin8ng the feature.

**Comment by mukulasija** [comment]
I would like to work on this issue, can you please assign it to me..

**Comment by kelson42** [comment]
@mukulasija OK, but please propose here a solution first to explain how you would do that.

**Comment by Saifuddin53** [comment]
@kelson42 I think #4211 resolves this issue as well, we can close this!
